crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -0,0 +1,123 @@
+use std::io;
+
+use ruff_db::system::{System, SystemPath, SystemPathBuf};
+
+/// Enumeration of the kinds of platform we might be running on.
+///
+/// For the purposes of discovering `site-packages` directories,
+/// there is only one distinction that matters here: whether or not
+/// we're running on windows.
+///
+/// Note: this is the platform of the *host machine*, which may be
+/// different from the *target platform for type checking*. (A user
+/// might be running red-knot on a Windows machine, but might
+/// reasonably ask us to type-check code assuming that the code runs
+/// on Linux.)
```

Comment:
Could you tell me more about how such a setup looks like? Would they create a windows venv on linux? How would that be done?

Reply:
You're on Windows (because it's the weekend, and you're away from the office), and you create a Windows virtual environment, but you pass `--platform=linux` to red-knot, because you know that in production your servers run on Linux, and you want to emulate the production environment when you're type-checking things locally.

The `--platform=linux` argument allows red-knot to statically infer that [branches like this](https://github.com/python/typeshed/blob/8a7f09e3511f3a1d04281c60167b8dcc3b78938b/stdlib/time.pyi#L13-L14) in stubs are always true for type-checking purposes, and that [branches like this](https://github.com/python/typeshed/blob/8a7f09e3511f3a1d04281c60167b8dcc3b78938b/stdlib/time.pyi#L28-L32) are always false. (This is [specified in PEP 484](https://peps.python.org/pep-0484/#version-and-platform-checking).)

However, when it comes to figuring out where the `site-packages` directory is in the virtual environment, the `--platform` argument is totally irrelevant. We need to figure out the actual platform of the host machine (which in this case is Windows), not the `--platform` argument that the user may or may not have passed.

Reply:
Thanks for the explanation but I think I'm now more confused than before :laughing: 

> However, when it comes to figuring out where the site-packages directory is in the virtual environment, the --platform argument is totally irrelevant.

Wouldn't that mean that we could just use `if cfg!(target_os = "windows") { .... }`?

> , I believe we'll need to have our tests depend on either Python or uv (we'll need to use one or the other to create the virtual environment). Do you have a preference?

I would just commit the created venv.

Reply:
> Wouldn't that mean that we could just use `if cfg!(target_os = "windows") { .... }`?

Yes, that also works!

> I would just commit the created venv.

Ah, that's a nice idea

Reply:
> Yes, that also works!

Oh okay. In that case, I would prefer to use `cfg` because we then only ship the relevant code in the released binary. It's also how we do this kind of branching in `uv`.

Reply:
Yes, that makes sense. I forgot that `cfg` was the Rusty way of doing this kind of thing. I'm working on the changes :-)